cmd/kube-controller-manager/app/core.go

```diff
@@ -143,14 +152,33 @@ func startNodeIpamController(ctx ControllerContext) (http.Handler, bool, error)
 		}
 	}
 
+	var nodeCIDRMaskSizeIPv4, nodeCIDRMaskSizeIPv6 int
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) {
```

Comment:
what happen if the feature gate is not enabled? 
how do I get `int(ctx.ComponentConfig.NodeIPAMController.NodeCIDRMaskSize),' and the cluster family?

Reply:
The set node cidr mask sizes only sets the IPv4 mask and IPv6 masks that we'll use. The `getNodeCIDRMaskSizes` returns a slice of node cidr mask sizes to be used based on cluster family.